src/components/LoadingScreen/LoadingScreen.js

```diff
@@ -70,102 +67,59 @@ class LoadingScreen extends PureComponent<Props, State> {
   }
 
   render() {
-    const { showLoadingScreen } = this.props;
-    /* 
-    // Todo: Create a better UI. Code commented for now.
-    const {statusText} = this.props;
-    const { showStatus } = this.state;
-    const ellipsis = statusText.length > 220 ? '...' : '';
-    const shortenedText =
-      statusText
-        .split('')
-        .splice(0, 220)
-        .join('') + ellipsis;
-    */
+    const { showLoadingScreen, statusText } = this.props;
+    const { reset } = this.state;
+    const { progress } = this;
+    const shortenedText = ellipsify(statusText, 100);
+
     return (
       <Window isVisible={showLoadingScreen}>
+        <FishSpinner src={guppyLoaderSrc} alt="Fish loader" />
         <ProgressBarWrapper>
```

Comment:
So I realized that this loading screen is also being used when you delete a project from disk, but that doesn't have progress or status. 

So I think we'll need to reformat it to conditionally render `ProgressBarWrapper` and `InfoWrapper` based on whether there's any status text or not. Something like...

```js
<Details shouldDisplay={progress !== null && statusText !== null}>
  <ProgressBarWrapper>
    <ProgressBar height={18} progress={0.75} reset={reset} />
  </ProgressBarWrapper>
  <InfoWrapper>
    <InfoText>{shortenedText}</InfoText>
    <InfoText>{0.75 * 100}%</InfoText>
  </InfoWrapper>
</Details>

....

const Details = styled.div`
  display: ${props => (props.shouldDisplay ? 'block' : 'none')};
  width: 100%;
`;
```

Or should there be a specific piece of state in `app-status.reducer.js` for this when you launch `REINSTALL_DEPENDENCIES_START`? Something like `displayProgressBar: true`? Not sure if this is overkill. Ping @AWolf81

Reply:
I think it's OK to conditional render based on `progress` and `statusText`. Just checking should be against 0 and empty string as that are the defaults. Note: `statusText` will be `Please wait...` as this is the initialState from reducer.

I think a boolean is not needed in `app-status.reducer` as we can get the same info if there would be the progress on state (see below).

I'm wondering if we should make the progress updating more reusable. The progress is pretty coupled to the text from the yarn output `1/4 ...` , `2/4 ...` etc.

I think we could add the progress to `app-status.reducer` and update it after updating the `statusText` in the following [line](https://github.com/joshwcomeau/guppy/blob/master/src/sagas/dependency.saga.js#L196) in `dependency.saga`. So it would be possible to display the progress with the value.

No need to add this now as this can be added later if we need the `LoadingScreen` in another place with progress bar.

Should we display the `statusText` centered below the spinner if there is no progress? I think that would be good to have as maybe we want to add more info to the current task e.g. `Please wait. Deleting in progress...`

Reply:
Yes, the split second could be a problem - not sure how to handle this. 

The idea behind `Please wait...` is that it will always fit and resetting to a consistent value is easy - just load the initial value. But you're right `null` would also work but it will require to set the status to get a text - if you forget to set it you won't get a text or you're getting an old status.

I like the `Please wait...` default as it's safer and it won't display a wrong or no text.